test/with_dummyserver/test_socketlevel.py

```diff
@@ -1491,6 +1492,44 @@ def socket_handler(listener: socket.socket) -> None:
                 pool.request("GET", "/", retries=False, timeout=LONG_TIMEOUT)
         assert server_closed.wait(LONG_TIMEOUT), "The socket was not terminated"
 
+    def test_openssl_overflow_on_read(self) -> None:
+        """
+        Ensure OpenSSL read doesn't fail with OverflowError when reading > (2**31)-1 bytes.
+        """
+
+        def socket_handler(listener: socket.socket) -> None:
+            sock = listener.accept()[0]
+            sock = ssl.wrap_socket(
+                sock,
+                server_side=True,
+                keyfile=DEFAULT_CERTS["keyfile"],
+                certfile=DEFAULT_CERTS["certfile"],
+                ca_certs=DEFAULT_CA,
+            )
+
+            sock.send(
+                b"HTTP/1.1 200 OK\r\n"
+                b"Content-type: text/plain\r\n"
+                b"Content-Length: 2147483648\r\n"
+                b"\r\n"
+            )
+            sock.close()
```

Comment:
This test might be failing due to CI running at different speeds than locally. Might require waiting for an `Event` object to run `sock.close()` as is being done in other places.